Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿using UnityEngine;
+using Reactor.Utilities.Attributes;
+using System;
+using System.Collections;
+using System.Linq;
+using Neuro.Recording;
+using Il2CppInterop.Runtime.Attributes;
+using BepInEx.Unity.IL2CPP.Utils;
+using Neuro.Utilities;
+using Neuro.Events;
+
+namespace Neuro.Impostor;
+
+[RegisterInIl2Cpp, ShipStatusComponent]
+public sealed class ImpostorHandler : MonoBehaviour
+{
+    public ImpostorHandler(IntPtr ptr) : base(ptr) { }
+
+    public static ImpostorHandler Instance { get; private set; }
+
+    private void Awake()
+    {
+        if (Instance)
+        {
+            NeuroUtilities.WarnDoubleSingletonInstance();
+            Destroy(this);
+            return;
+        }
+
+        Instance = this;
+    }
+
+    [HideFromIl2Cpp]
+    public void HandleVent(int id)
+    {
+        Vent vent = ShipStatus.Instance.AllVents.First(v => v.Id == id);
+        this.StartCoroutine(CoStartVentOut(vent));
+    }
+
+    [HideFromIl2Cpp]
+    private IEnumerator CoStartVentOut(Vent original)
+    {
+        Vent[] possibleVents = GetAvailableNearbyVents(original);
+        Vent current = possibleVents[^1];
+        yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
+        if (!original.TryMoveToVent(current, out string error))
+        {
+            Error($"Failed to move to vent {current.Id}, reason: {error}");
+        }
+        while (true)
+        {
+            bool playerSpotted = false;
+            yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
+            Frame now = Frame.Now(true);
+            foreach (var player in now.OtherPlayers.LastSeenPlayers)
+            {           
+                // if we see a crewmate in our radius, try a different vent
+                if (player.IsVisible && !now.Header.OtherImpostors.Contains(player.Id))
```

Comment:
I would suggest pathfinding to players as well to check for a minimum distance. Even though that's """cheating""", it will lead to better results as an impostor and not lead to situations where neuro travels to a vent where a player is there, waits between 0.8 and 1.2 seconds as you specified above and during that time the player moves out of vision, then neuro vents out and the player double-backs. etc